src/main/java/ulcrs/data/DataStore.java

```diff
@@ -0,0 +1,156 @@
+package ulcrs.data;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import ulcrs.models.course.Course;
+import ulcrs.models.shift.Shift;
+import ulcrs.models.tutor.Tutor;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.DayOfWeek;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataStore {
+
+    private List<Tutor> tutors;
+    private List<Course> courses;
+    private List<Shift> shifts;
+    private LocalDateTime timeFetched = LocalDateTime.MIN;
+
+    private static DataStore dataStore;
+
+    private static DataStore getInstance() {
+        if (dataStore == null) {
+            dataStore = new DataStore();
+        }
+        return dataStore;
+    }
+
+
+    /**
+     * Get the tutor of the specified id
+     * @param id - id of the tutor
+     * @return Tutor - the Tutor with the specified id, if it exists. Otherwise, null.
+     */
+    public static Tutor getTutor(int id) {
+        fetchIfRequired(getInstance().tutors);
+        return getInstance().tutors.stream()
+                .filter(tutor -> tutor.getId() == id)
+                .findFirst()
+                .orElse(null);
+    }
+
+    /**
+     * Get the course of the specified id
+     * @param id - id of the course
+     * @return Course - the Course with the specified id, if it exists. Otherwise, null.
+     */
+    public static Course getCourse(int id) {
+        fetchIfRequired(getInstance().courses);
+        return getInstance().courses.stream()
+                .filter(course -> course.getId() == id)
+                .findFirst()
+                .orElse(null);
+    }
+
+    /**
+     * Get the shift of the specified id
+     * @param id - id of the shift
+     * @return Shift - the shift with the specified id, if it exists. Otherwise, null.
+     */
+    public static Shift getShift(int id) {
+        fetchIfRequired(getInstance().shifts);

```

Comment:
nitpick: space

Reply:
What do you mean? I'm not sure I see a space

Reply:
Rip, clicked the wrong one. It's a bit further down, line 98. And tab on line 109.

Reply:
Should be updated